### ISSUE TITLE
Spurious "Unable to find credentials" error logs in `sbt update` with artifactory

### ISSUE THREAD

**Issue by alexklibisz** [sbt/sbt#7141]
## steps

1. Set credentials: `credentials += Credentials("Artifactory Realm", "artifactory.company.tld", "valid username", "valid password")`
2. Set externalResolvers: `externalResolvers := Seq(Resolver.defaultLocal, "Company Artifacts" at "artifactory.company.tld")
3. Run `sbt -client update`

## problem

The `sbt -client update` step often prints a stack of spurious error logs like this:

<img width="727" alt="image" src="https://user-images.githubusercontent.com/8015228/214932315-c64c35a2-9cd2-4401-82f3-34593cb2e736.png">

But then it successfully completes the command.

The greyed out boxes are the "artifactory.company.tld"

## expectation

Don't print error logs when the command is correct.

## notes

* This doesn't always happen
* The frequency of error logs seems random

**Comment by alexklibisz** [comment]
Looks like the error might be coming from here: https://github.com/sbt/librarymanagement/blob/54011d5e74335bab229bc3f6532b6a1ca6d0d5cd/ivy/src/main/scala/sbt/internal/librarymanagement/ivyint/ErrorMessageAuthenticator.scala#L125

**Comment by alexklibisz** [comment]
One thing I don't understand is that this error/warning message is coming from code that's related to Ivy. I thought that SBT has been using coursier for a while now. Maybe it's using coursier to resolve from an Ivy repository?

**Comment by SethTisue** [comment]
> One thing I don't understand is that this error/warning message is coming from code that's related to Ivy. I thought that SBT has been using coursier for a while now. Maybe it's using coursier to resolve from an Ivy repository?

You don't have `useCoursier := false` set, do you?

**Comment by alexklibisz** [comment]
Correct, I haven't set `useCoursier := false`.

**Comment by SethTisue** [comment]
I agree it seems downright weird. I'm afraid I don't know this part of the codebase and I don't have any special insight. A couple general suggestions though, one would be to try running with `-debug` to see if the debug logs give any additional clues. Another is to attach a debugger to the sbt process and set a breakpoint at the line where the error is generated, to see what stack frames led us there.

**Comment by mzuehlke** [comment]
You haven't specified wich sbt version you are using.
The usage of ivy could come from a sbt plugin. Which plugins, in which versions are you using?

**Comment by normana400** [comment]
I have the issue with sbt 1.11.0

**Comment by eed3si9n** [comment]
If someone wants to track this down, they are welcome, but sbt project probably won't pursue fixing log-only issue that I won't be able to easily reproduce.
